cmd/ftl/cmd_schema_import.go

```diff
@@ -0,0 +1,220 @@
+		"The resulting Go file should represent every object as its own Go struct, written as `type...`. "+
+		"Everything in the provided input that is not a type or object declaration must be ignored."+
+		"No fields should be modified or added—the result must be an exact translation of the provided input, retaining case."+
+		"The result should reside in a package named `ftl`."+
+		"Your output should only include the resultant Go types with no additional functions."+
+		"Please provide output that is only the resultant Go file, with no additional text or explanation. The initial string to translate is this: %s",
+		input,
+	), nil
+}
+
+func (i *importGoCmd) Run(ctx context.Context, parent *schemaImportCmd) error {
+	err := parent.setup(ctx)
+	if err != nil {
+		return err
+	}
+
+	prompt, err := i.getPrompt()
+	if err != nil {
+		return err
+	}
+
+	err = query(ctx, prompt)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *importKotlinCmd) Run(ctx context.Context, parent *schemaImportCmd) error {
+	err := parent.setup(ctx)
+	if err != nil {
+		return err
+	}
+
+	prompt, err := i.getPrompt()
+	if err != nil {
+		return err
+	}
+
+	err = query(ctx, prompt)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func query(ctx context.Context, prompt string) error {
+	logger := log.FromContext(ctx)
+
+	logger.Infof("The import schema command relies on the %s AI chat model to translate schemas from other "+
+		"languages into FTL compliant objects in the specified language. Output may vary and results should be inspected "+
+		"for correctness. It is suggested that if the results are not satisfactory, you try again.\n\n", ollamaModel)
+
+	llm, err := ollama.New(ollama.WithModel("llama2"))
+	if err != nil {
+		logger.Errorf(err, "failed to call ollama")
+	}
+
+	completion, err := llm.Call(ctx, fmt.Sprintf("Human: %s \nAssistant:", prompt),
+		llms.WithTemperature(0.8),
+		llms.WithStreamingFunc(func(ctx context.Context, chunk []byte) error {
+			fmt.Print(string(chunk))
+			return nil
+		}),
+	)
+	if err != nil {
+		logger.Errorf(err, "failed to call ollama")
+	}
+
+	_ = completion
+	return nil
+}
+
+func (s *schemaImportCmd) setup(ctx context.Context) error {
+	logger := log.FromContext(ctx)
+
+	nameFlag := fmt.Sprintf("name=^/%s$", ollamaContainerName)
+	output, err := exec.Capture(ctx, ".", "docker", "ps", "-a", "--filter", nameFlag, "--format", "{{.Names}}")
+	if err != nil {
+		logger.Errorf(err, "%s", output)
+		return err
+	}
+
+	port := fmt.Sprintf("%d", s.OllamaPort)
+
+	if len(output) == 0 {
+		logger.Infof("Creating docker container '%s' for ollama", ollamaContainerName)
+
+		// check if port s.OllamaPort is already in use
+		_, err := exec.Capture(ctx, ".", "sh", "-c", fmt.Sprintf("lsof -i:%d", s.OllamaPort))
+		if err == nil {
+			return fmt.Errorf("port %d is already in use", s.OllamaPort)
+		}
```

Comment:
I realise you reused an existing pattern in FTL here (which I must have missed in a previous review, my bad), but it's better not to rely on a minimal (or ideally zero) set of external executables because it requires end users to do a bunch of setup. For this case (and the other one in `cmd_serve.go`) we should instead try to bind to the port using `net.Listen()` and immediately close it if it succeeds.

```suggestion
		l, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", o.OllamaPort))
		if err == nil {
			return fmt.Errorf("port %d is already in use", s.OllamaPort)
		}
		_ = l.Close()
```